includes/class-wc-stripe-order.php

```diff
@@ -0,0 +1,740 @@
+	 *
+	 * @return bool Whether the payment is awaiting action.
+	 */
+	public function payment_awaiting_action() {
+		return wc_string_to_bool( $this->get_meta( '_stripe_payment_awaiting_action' ) );
+	}
+
+	/**
+	 * Removes the metadata from the order that was used to indicate that the payment was awaiting action.
+	 *
+	 * @param bool     $save  Whether to save the order after removing the metadata.
+	 *
+	 * @return void
+	 */
+	public function remove_payment_awaiting_action( $save = true ) {
+		$this->delete_meta_data( '_stripe_payment_awaiting_action' );
+
+		if ( $save ) {
+			$this->save();
+		}
+	}
+
+	/**
+	 * Set the preferred card brand.
+	 *
+	 * @param $brand string The brand to set.
+	 * @return void
+	 */
+	public function set_card_brand( $brand ) {
+		$this->update_meta_data( '_stripe_card_brand', $brand );
+	}
+
+	/**
+	 * Get the preferred card brand.
+	 *
+	 * @return string
+	 */
+	public function get_card_brand() {
+		return $this->get_meta( '_stripe_card_brand' );
+	}
+
+	/**
+	 * Locks an order for refund processing for 5 minutes.
+	 *
+	 * @return bool A flag that indicates whether the order is already locked.
+	 */
+	public function lock_refund() {
+		$this->read_meta_data( true );
+
+		$existing_lock = $this->get_lock_refund();
+
+		if ( $existing_lock ) {
+			$expiration = (int) $existing_lock;
+
+			// If the lock is still active, return true.
+			if ( time() <= $expiration ) {
+				return true;
+			}
+		}
+
+		$new_lock = time() + 5 * MINUTE_IN_SECONDS;
+
+		$this->set_lock_refund( $new_lock );
+		$this->save_meta_data();
+
+		return false;
+	}
+
+	/**
+	 * Unlocks an order for processing refund.
+	 */
+	public function unlock_refund() {
+		$this->delete_meta_data( '_stripe_lock_refund' );
+		$this->save_meta_data();
+	}
+
+	/**
+	 * Set the lock refund time.
+	 *
+	 * @param $time int The time to set.
+	 * @return void
+	 */
+	public function set_lock_refund( $time ) {
+		$this->update_meta_data( '_stripe_lock_refund', $time );
+	}
+
+	/**
+	 * Get the lock refund time.
+	 *
+	 * @return int
+	 */
+	public function get_lock_refund() {
+		return $this->get_meta( '_stripe_lock_refund' );
+	}
+
+	/**
+	 * Set the setup intent.
+	 *
+	 * @param $value string The value to set.
+	 * @return void
+	 */
+	public function set_setup_intent( $value ) {
+		$this->update_meta_data( '_stripe_setup_intent', $value );
+	}
+
+	/**
+	 * Get the setup intent.
+	 *
+	 * @return string
+	 */
+	public function get_setup_intent() {
+		return $this->get_meta( '_stripe_setup_intent' );
+	}
+
+	/**
+	 * Set the UPE redirect processed flag.
+	 *
+	 * @param $value bool The value to set.
+	 * @return void
+	 */
+	public function set_upe_redirect_processed( $value ) {
+		$this->update_meta_data( '_stripe_upe_redirect_processed', $value );
+	}
+
+	/**
+	 * Whether the UPE redirect has been processed.
+	 *
+	 * @return bool The value of the flag.
+	 */
+	public function upe_redirect_processed() {
```

Comment:
For consistency, should be `is_ upe_redirect_processed ` or `get_ upe_redirect_processed`

Reply:
Addressed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3654/commits/1dc95df6d048882b482e40171c6af42b7ce32796